includes/WcfceMailProcessor.inc

```diff
@@ -0,0 +1,223 @@
+<?php
+
+/**
+ * @file
+ * Webform submission mail processor.
+ */
+
+class WcfceMailProcessor {
+
+  /**
+   * Email data (comes from hook_mail_alter()).
+   *
+   * @var array
+   */
+  private $message;
+
+  /**
+   * WcfceMailProcessor constructor.
+   *
+   * @param array $message
+   *   Email data (comes from hook_mail_alter()).
+   */
+  public function __construct(array $message) {
+    $this->message = $message;
+  }
+
+  /**
+   * File email as activity for all cases related to the webform.
+   *
+   * @return array|false
+   *   Entity data of the last activity added, empty array or FALSE
+   *   in case of error.
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  public function createActivity() {
+    if (!$this->isMessageValid()) {
+      return FALSE;
+    }
+
+    if (!civicrm_initialize()) {
+      return FALSE;
+    }
+
+    $assignee_id = $this->getContactIdByEmail($this->message['to']);
+    $creator_id  = $this->getContactIdByEmail($this->message['from']);
```

Comment:
Can the activity creator be gotten from the `From` email address? I think each site is configured to have some default from email addresses and this can be picked from there? (Can you verify this?)
Was thinking if it will be better to get the creator_id from the civicrm contact ID of the logged in user.